src/aind_data_transfer/jobs/zarr_upload_job.py

```diff
@@ -203,22 +205,98 @@ def _upload_zarr(self) -> None:
+        # log_dict = read_imaging_log(log_file)
+
+        # #if any of the values of log_dict are None, then get it from schema_log
+        # if any(v is None for v in log_dict.values()):
+        #     self._instance_logger.info('Some values in imaging_log.log are None. Reading schema_log.log')
+        #     log_file =  self._data_src_dir.joinpath('schema_log.log')
+        #     log_dict = {}
+        #     log_dict = read_schema_log_file(log_file)
+        #     log_dict['data_src_dir'] = (self._data_src_dir.as_posix())
+        #     log_dict['config_toml'] = toml_dict
+        #     self._instance_logger.info('Finished reading schema_log.log')
+        #     acq_json = schema_log_to_acq_json(log_dict)
+        #     self._instance_logger.info('Finished converting schema_log.log to acq json')
+        # else:
+        #     log_dict['data_src_dir'] = (self._data_src_dir.as_posix())
+        #     log_dict['config_toml'] = toml_dict
+        #     # convert to acq json
+        #     acq_json = log_to_acq_json(log_dict)
+        # # convert to acq json
+        # acq_json_path = self._data_src_dir.joinpath('acquisition.json')
+
+        # try:
+        #     write_acq_json(acq_json, acq_json_path)
+        #     self._instance_logger.info('Finished writing acq json')
+        # except Exception as e:
+        #     self._instance_logger.error(f"Failed to write acquisition.json: {e}")
+
+        # # convert acq json to xml
+        # is_zarr = True
+        # condition = "channel=='405'"
+        # acq_xml = acq_json_to_xml(acq_json, log_dict, self.job_configs.s3_prefix + f'/{self._modality_config.modality.value.abbreviation}.ome.zarr', is_zarr, condition)  # needs relative path to zarr file (as seen by code ocean)
+
+        # # write xml to file
+        # xml_file_path = self._data_src_dir.joinpath('Camera_405.xml')  #
+        # write_xml(acq_xml, xml_file_path)
+
         log_file = self._data_src_dir.joinpath('imaging_log.log')
+        acq_file = self._data_src_dir.joinpath('dispim_acquisition.json')
+
+
         toml_dict = read_toml(self._data_src_dir.joinpath('config.toml'))
+        
 
         # read log file into dict
-        log_dict = read_imaging_log(log_file)
-        log_dict['data_src_dir'] = (self._data_src_dir.as_posix())
-        log_dict['config_toml'] = toml_dict
+        if not log_file.exists():
+            log_dict = read_imaging_log(log_file)
+            log_dict['data_src_dir'] = (self._data_src_dir.as_posix())
+            log_dict['config_toml'] = toml_dict
+        
+        else: 
+            log_dict = {'imaging_log_file': None} #set this to none to read schema_log
+
+
+        if acq_file.exists():
+            acq_json = read_dispim_aquisition(acq_file)
+            
+
+        #if any of the values of log_dict are None, then get it from schema_log
+        elif any(v is None for v in log_dict.values()):
+            print('Some values in imaging_log.log are None. Reading schema_log.log')
```

Comment:
can we replace the print statements with the instance logger?

Reply:
Yes, good point. Will do!